### ISSUE TITLE
Conflicts between `no-prototype-builtins` and `no-useless-call`

### ISSUE THREAD

**Issue by fisker** [eslint/eslint#14659]
**Tell us about your environment**

* **ESLint Version:** 7.27.0
* **Node Version:** 16
* **npm Version:** 6
* **Operating System:** windows 

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

default(espree)

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
// I don't think this is useful
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I want check a property is Object's own property,

```js
Object.prototype.hasOwnProperty(property);
```

`no-prototype-builtins` reports an error.

```js
Object.prototype.hasOwnProperty.call(Object.prototype, property);
```

`no-useless-call` reports an error.

**What did you expect to happen?**

`no-useless-call` rule should ignore `Object.prototype.hasOwnProperty.call(Object.prototype, ...)`
 

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

N/A

**Steps to reproduce this issue:**

N/A

**Are you willing to submit a pull request to fix this bug?**

No

**Comment by ljharb** [comment]
Why would you want to check that Object.prototype has an own property?

**Comment by fisker** [comment]
Here is the case, https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1330/files

I want ignore `objectLiteral.method` but not `objectLiteral.propertyIsEnumerable`

**Comment by fisker** [comment]
Here is another PR, https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1296/files#diff-e409151331375f48a019cd347434ee0445bac8841d55bc8fa935e54e299d1ff1R107 doing the same thing.

//cc bmish

**Comment by ljharb** [comment]
This seems like a very obscure edge case to me, and using an override comment seems fine. Personally, I'd go with `Object.prototype.hasOwnProperty.call(Object.prototype, property)` since that's more explicit.

You may also be interested in the stage 3 proposal, https://npmjs.com/object.hasown, which allows you to `Object.hasOwn(Object.prototype, property)` and avoid both of these concerns.

**Comment by fisker** [comment]
We already have [`unicorn/prefer-object-has-own`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-object-has-own.md) rule, haha.

**Comment by fisker** [comment]
I updated expect to

`no-useless-call` rule should ignore `Object.prototype.hasOwnProperty.call(Object.prototype, ...)`

**Comment by nzakas** [comment]
Yeah, I also agree that this is more of an edge case and I can’t see a generic way to solve the conflict that doesn’t cause other problems. When there are as many core rules as ESLint has, it just gets difficult to make them all play together in every configuration.

**Comment by mdjermanovic** [comment]
I also think this is a very edge case, and a good case for an `eslint-disable-*` comment.

**Comment by nzakas** [comment]
Consensus on the team is to not make any changes here, so closing.
